### ISSUE TITLE
Cover expenses on the website

### ISSUE THREAD

**Issue by levinli303** [celestiamobile/celestia.mobi#39]
celestia.mobi is unsurprisingly not free. The expenses are in two parts:

1. the domain celestia.mobi, I purchased the domain on Tencent Cloud and it costs about 20 USD per year
2. server and other infrastructure, the server is on Azure with a pay as you go billing model and costs about 1320 USD per year

I've been paying for 1, and 2 has been free at the moment for some reason but in time I will have to pay to keep it running. I have three options in mind:

1. merge with celestia.space (I personally do not like it because a lot of things are tied with celestia.mobi domain already, and the two sites are built very differently)
2. seek sponsors like fosshost to provide server/domain (I do not know any other than fosshost)
3. ads in celestia (for example add-on pages, I'm not sure if it will cover all the expenses though)

**Comment by GurrenLagannTSS** [comment]
Personally i like option 2 more, lacking any other options. A cool suggestion would be to add a donation page to help with the server costs.

**Comment by TheLostProbe** [comment]
this is a scenario in which a celestia patreon could make sense. but for now i think just go with fosshost. i really dont think celestia should _ever_ have ads, and merging mobi with celestia.space would be a very tough undertaking

**Comment by SevenSpheres** [comment]
Hosting through something like Fosshost would be the best option. In this case, instead of a donation page, there could be a "Donate to Fosshost" link.

**Comment by Anthony-B-Russo10** [comment]
> this is a scenario in which a celestia patreon could make sense. but for now i think just go with fosshost. i really dont think celestia should _ever_ have ads, and merging mobi with celestia.space would be a very tough undertaking

Plus if it did merge, I don't think Levin would have that much control of parts of mobi as OneTwoThree hosts the main site.

**Comment by TTarrants** [comment]
I may be able to pay the $1320 cost of the mobi site hosting for a year, perhaps further.
-- Thomas (Sirius_Alpha).

**Comment by levinli303** [comment]
> I don't think Levin would have that much control of parts of mobi as OneTwoThree hosts the main site.

I have access to celestia.space server although I have not logged in for a long time.

> I may be able to pay the $1320 cost of the mobi site hosting for a year, perhaps further.

That's very kind of you. At the moment it is not necessary, as this part of cost is covered. When the time comes it needs of be paid for and we have no other solution, I'll think about it.

For the Fosshost (or other sponsor) solution, i do think having donate button in web pages will be good. But a good questions would be, will they sponsor it?

**Comment by TheLostProbe** [comment]
well considering that fosshost already hosts celestia.space, i would be very surprised to see them reject the application for celestia.mobi. problem is, [fosshost has temporarily stopped accepting applications](https://fosshost.org/news/suspended-applications)

**Comment by levinli303** [comment]
> [fosshost has temporarily stopped accepting applications](https://fosshost.org/news/suspended-applications)

that's a good point. then I think we can go ahead with patreon. I do not know what we can expect but if we receive more money than the website requires I can use it on marketing Celestia, etc.

**Comment by TheLostProbe** [comment]
great, then i guess we should add a patreon link to celestia.space and celestia.mobi, and also make a topic on the forums. on the topic of marketing celestia, what do you think about creating a celestia youtube channel?

**Comment by levinli303** [comment]
as for patreon links, I still need to know how it works. if anyone are willing to help out, be my guest.

as for marketing, I don't have very concrete ideas. Some App Stores have ads campaign where they put your app on top in certain search results or other places. that is what I have in mind where money can be spent. creating a YouTube channel should be free.

**Comment by TheLostProbe** [comment]
patreon links could probably be done by just grabbing a "Become a Patron" or "Support on Patreon" image off the internet ([example](https://stonation.life/wp-content/uploads/2021/04/patreon-button.png)) and then doing something like:
`<a href="https://www.patreon.com/Celestia"><img src="images/patreon-button.png" /></a>`
although i think it is possible to use `href` in `<img>` elements but i dont remember correctly. if so, there is no need for an `<a>` element

regarding the youtube channel idea, i know that creating them is free, but im just not very sure who here would be willing to dedicate their time to creating and editing videos to upload on the channel

**Comment by Anthony-B-Russo10** [comment]
YouTube is free, though if the videos become monetized the project could get an extra cash flow from those videos. Plus we could also use someone who dedicates their time to such a channel who also has really good video editing skills.

**Comment by levinli303** [comment]
I created a patreon page, https://www.patreon.com/celestiamobi . but it is far from complete, like how to get paid, etc.

**Comment by levinli303** [comment]
https://www.patreon.com/celestiamobi should be good now, I have set up membership and payment, just need to add an icon to celestia.mobi...

**Comment by TheLostProbe** [comment]
> doing something like:
> `<a href="https://www.patreon.com/Celestia"><img src="images/patreon-button.png" /></a>`

actually, we can just use the patreon icon from fontawesome and embedding a `href` into it
https://fontawesome.com/icons/patreon?s=solid&f=brands

**Comment by levinli303** [comment]
> actually, we can just use the patreon icon from fontawesome and embed a `href` into it https://fontawesome.com/icons/patreon?s=solid&f=brands

Not gonna embed an entire font for one glyph. I'll just create a simple vector image for this purpose.

**Comment by TheLostProbe** [comment]
no need to pay for fontawesome to use it, you can just download the single svg here:
![image](https://user-images.githubusercontent.com/67564416/203172685-b16c891a-bbd7-490b-b765-1f4461f2bece.png)

**Comment by levinli303** [comment]
> no need to pay for fontawesome to use it, you can just download the single svg here:
> ![image](https://user-images.githubusercontent.com/67564416/203172685-b16c891a-bbd7-490b-b765-1f4461f2bece.png)

it's not really about paying. I just do not really like this icon.

**Comment by TheLostProbe** [comment]
okay, then what kind of icon do you have in mind?

**Comment by DaveBowman2001** [comment]
> merge with celestia.space (I personally do not like it because a lot of things are tied with celestia.mobi domain already, and the two sites are built very differently)

Well as of today, fosshost is no longer an option ever due to [Hosting provider shutting down, new hosting urgently needed](https://celestia.space/forum/viewtopic.php?f=17&t=23479) - @SevenSpheres, Celestia Forums

**Comment by levinli303** [comment]
we can host two domains on a same server. but the configuration of the server should be improved.

**Comment by TheLostProbe** [comment]
great idea! i would recommend doing something akin to this for the celestia logo, but with colour on the "C" and a different font since this is the segoe ui font:
![logo](https://user-images.githubusercontent.com/67564416/206606407-a8152108-2f65-4d2e-b07f-53311ba25b32.png)

the "elestia" part of it may be too small, though

**Comment by levinli303** [comment]
Two drafts, I'll probably use the first one, a different font will be needed.

<img width="667" alt="Screenshot 2022-12-12 at 21 42 26" src="https://user-images.githubusercontent.com/17924287/207059909-836b46e6-3afa-4119-ac8a-d2b558357696.png">
<img width="708" alt="Screenshot 2022-12-12 at 21 42 53" src="https://user-images.githubusercontent.com/17924287/207059927-18f1792c-b331-4854-b89a-c3c548d9cb3c.png">

**Comment by levinli303** [comment]
I figured that eventually I want to use only the icon and have the text in HTML, I used the color from patreon website as the background. An example page.

<img width="870" alt="Screenshot 2022-12-15 at 18 33 02" src="https://user-images.githubusercontent.com/17924287/207837110-ba547453-366b-440f-bac0-f05246cdfa88.png">

**Comment by levinli303** [comment]
let me know what you think. The CSS I used here is

```
 .patreon-icon {
       height:calc(100% - 16px);
       margin-left: 8px;
       margin-right: 4px;
   }

   .patreon-text {
       color: white;
       font-size: 22px;
       font-weight: 600;
       margin-right: 10px;
   }

   .patreon-banner {
       display:inline-flex;
       align-items:center;
       background-color: #eb5354;
       height: 48px;
       max-width: calc(100%);
       border-radius: 12px;
       -webkit-user-select:none;
       -moz-user-select:none;
       -ms-user-select:none;
       user-select:none;
   }

   .patreon-banner:hover {
       text-decoration:none
   }

   .patreon-banner:visited {
       text-decoration:none
   }
```

HTML

```
<p><a class="patreon-banner" href="#"><img class="patreon-icon" src="https://cdn.astroweather.cn/celestia/v1/img/celestia-vectorized-icon.svg"><p class="patreon-text">Support Us on Patreon</p></a></p>
```

**Comment by TheLostProbe** [comment]
it looks really great! i think it would look better on the right side, since there is a lot of free space there and filling that space in looks better:
![image](https://user-images.githubusercontent.com/67564416/208008703-2dd14129-7db4-444e-9184-b7258d84ce90.png)

**Comment by levinli303** [comment]
> it looks really great! i think it would look better on the right side, since there is a lot of free space there and filling that space in looks better:
> ![image](https://user-images.githubusercontent.com/67564416/208008703-2dd14129-7db4-444e-9184-b7258d84ce90.png)

on mobile there is no such text on the left. and not much space

**Comment by levinli303** [comment]
the patreon button now is live on all add-on/article pages

**Comment by TheLostProbe** [comment]
> and not much space

true, it looks better on mobile because the width is much less than on desktop, so the free space on the right doesnt look as odd

btw i was wondering if this should be linked on the main site, and also if we should announce this in the news page on both celestia.mobi and celestia.space

**Comment by levinli303** [comment]
> btw i was wondering if this should be linked on the main site, and also if we should announce this in the news page on both celestia.mobi and celestia.space

we should not add this as news on celestia.space as celestia.mobi operates independently. however a news page on celestia.mobi would be good

**Comment by levinli303** [comment]
im planning to remove the Patreon button as this has not turned out to be effective.

**Comment by levinli303** [comment]
> im planning to remove the Patreon button as this has not turned out to be effective.

It has been largely unsuccessful, i do think that most users don't have or want to create a patreon account just for this occasion. Another option would be tipping through in-app purchase but that would require me to write code for each platform (and possibly for each app stores) to support in-app purchase all all frontends.

**Comment by Anthony-B-Russo10** [comment]
One way to make money is with a monetized YouTube account, but we would need someone who can run it.

**Comment by levinli303** [comment]
> One way to make money is with a monetized YouTube account, but we would need someone who can run it.

i really doubt that one can monetize on YouTube without a significant number of views, but i could be wrong.

On a side note, I have received several emails from people who wanted to promote Celestia on their TikTok/YouTube about astronomy. I declined their offers because I suppose they would ask for money for promotion.

**Comment by Anthony-B-Russo10** [comment]
Yeah, you would have to get into the YouTube partner program and meet certain thresholds (which for base level monetization is 500 subs, 3 public videos in the past 90 days, and 3000 watch hours on long form videos in the past 365 days). Though it can be worth a shot as any way of getting Celestia out into the public space helps.
On paid promotions, you would have to pay the content creators for them to promote the program.

**Comment by levinli303** [comment]
> Yeah, you would have to get into the YouTube partner program and meet certain thresholds (which for base level monetization is 500 subs, 3 public videos in the past 90 days, and 3000 watch hours on long form videos in the past 365 days). Though it can be worth a shot as any way of getting Celestia out into the public space helps. On paid promotions, you would have to pay the content creators for them to promote the program.

well we would still need people who run these accounts.

**Comment by TheLostProbe** [comment]
> well we would still need people who run these accounts.

I can run a YouTube account, but I'm not sure about recording the videos and editing them, since my computer's not very powerful. I do know that zelario from the discord server has editing experience

**Comment by levinli303** [comment]
> > well we would still need people who run these accounts.
> 
> I can run a YouTube account, but I'm not sure about recording the videos and editing them, since my computer's not very powerful. I do know that zelario from the discord server has editing experience

i think good ideas are more important than skills 😉. If we have the right folks, i will open accounts for celestia.mobi officially and advertise them in app too.

**Comment by Anthony-B-Russo10** [comment]
> > well we would still need people who run these accounts.
> 
> I can run a YouTube account, but I'm not sure about recording the videos and editing them, since my computer's not very powerful. I do know that zelario from the discord server has editing experience

Celestia has an inbuilt video recorder, but the only way to get good quality videos from it is to have the video uncompressed which results in videos larger then a gigabyte commonly. So it is best to try and find a screen recorder that is much lighter then Celestia's.
In terms of video editors, the one I have experience with (that isn't Windows Movie Maker) is OpenShot which has a minimum spec requirement that I can run it on a laptop that is from pre-2017, so you should be fine with it (albeit it not being the absolute best video editor out there).

**Comment by levinli303** [comment]
i do wonder if it is possible to do editing on phones if we go for short videos. let's say if we want to make YouTube Shorts, would apps on the phones suffice for editing?

**Comment by TheLostProbe** [comment]
there are plenty of capable video editing apps for android and ios so yes

**Comment by TheLostProbe** [comment]
thinking about creating a Celestia YouTube channel some time soon, potentially a TikTok as well but I am not sure
we should also get a Facebook account at some point (not too sure about Instagram just yet, and I don't know who would want to manage either of them), and maybe even a Mastodon and/or Bluesky account

**Comment by DaveBowman2001** [comment]
I can help with the Facebook/Instagram accounts

**Comment by levinli303** [comment]
created a separate issue for promoting Celestia #73 .

Let's keep this issue for monetization.

**Comment by levinli303** [comment]
spent another $22 on extending the expiry of celestia.mobi domain 🙃

**Comment by levinli303** [comment]
i'm planning to remove patreon soon.

**Comment by levinli303** [comment]
I'm experimenting with ads on celestia.mobi in browsers now. No ads are shown in app.

Use this link to access ad-less version in browsers https://celestia.mobi/resources/categories?showAds=false
